File: lib/src/main/java/io/github/goto1134/structurizr/export/d2/model/D2Property.java
package io.github.goto1134.structurizr.export.d2.model;

import com.structurizr.export.IndentingWriter;

/**
 * @param <K> – D2 object property name
 * @param <V> - D2 object property value
 */
public class D2Property<K, V> {

    protected final K keyword;
    protected final V value;

    public D2Property(K keyword, V value) {
        this.keyword = keyword;
        this.value = value;
    }

    public void write(IndentingWriter writer) {
        writer.writeLine(String.format("%s: %s", keyword, value));
    }

    public K getKeyword() {
        return keyword;
    }
}

File: lib/src/main/kotlin/io/github/goto1134/structurizr/export/d2/model/D2Property.kt
package io.github.goto1134.structurizr.export.d2.model

import com.structurizr.export.IndentingWriter

/**
 * @param <K> – D2 object property name
 * @param <V> - D2 object property value
</V></K> */
open class D2Property<K, V>(@JvmField val keyword: K, @JvmField protected val value: V) {
    open fun write(writer: IndentingWriter) = writer.writeLine("$keyword: $value")
}
